File: src/cn/vove7/plugin/rest/filetype/RestLexerAdapter.java
package cn.vove7.plugin.rest.filetype;

import com.intellij.lexer.FlexAdapter;
import com.intellij.lexer.MergingLexerAdapter;
import com.intellij.psi.tree.TokenSet;
import cn.vove7.plugin.rest.psi.RestTypes;

/**
 *
 */
public class RestLexerAdapter extends MergingLexerAdapter {
    public RestLexerAdapter() {
        super(new FlexAdapter(new RestLexer(null)), TokenSet.create(RestTypes.BAD_CHARACTER));
    }
}

File: src/main/kotlin/cn/vove7/plugin/rest/filetype/RestLexerAdapter.kt
package cn.vove7.plugin.rest.filetype

import com.intellij.lexer.MergingLexerAdapter
import com.intellij.lexer.FlexAdapter
import cn.vove7.plugin.rest.filetype.RestLexer
import com.intellij.psi.tree.TokenSet
import cn.vove7.plugin.rest.psi.RestTypes

/**
 *
 */
class RestLexerAdapter :
    MergingLexerAdapter(FlexAdapter(RestLexer(null)), TokenSet.create(RestTypes.BAD_CHARACTER))